crates/lib/src/compiler/quilc.rs

```diff
@@ -307,13 +307,26 @@ struct QuilToNativeQuilResponse {
     /// The compiled program
     quil: String,
     /// Metadata about the compiled program
+    #[serde(default)]
     metadata: Option<NativeQuilMetadata>,
 }
 
+#[allow(unused_qualifications)]
+fn deserialize_none_as_default<'de, D, T>(deserializer: D) -> Result<T, D::Error>
+where
+    D: Deserializer<'de>,
+    T: Deserialize<'de> + std::default::Default,
+{
+    let opt = Option::deserialize(deserializer)?;
+    Ok(opt.unwrap_or_default())
+}
+
 /// Metadata about a program compiled to native quil.
 #[derive(Clone, Deserialize, Debug, PartialEq, PartialOrd)]
 pub struct NativeQuilMetadata {
     /// Output qubit index relabeling due to SWAP insertion.
+    // #[serde(deserialize_with = "deserialize_optional_unit_or_vec_as_vec")]
+    #[serde(deserialize_with = "deserialize_none_as_default")]
```

Comment:
We could also change the type to `Option<Vec<u64>>`, but it would would make the API annoyingly clunky to use. If I understand correctly, `quilc` only returns `None` here if no qubits were used in the program. Most users are going to care about programs that do use qubits, and it would be annoying to have to check for `None` each time. I assume that's why [this logic exists in the rpcq package](https://github.com/rigetti/rpcq/blob/master/rpcq/messages.py#L237) as well.